libs/stats/odc/stats/_gm_ls_bitmask.py

```diff
@@ -0,0 +1,154 @@
+        self.resampling = resampling
+        self.bands = bands
+        self.work_chunks = work_chunks
+        self.renames = aux_names
+        self.aux_bands = list(aux_names.values())
+
+        if bands is None:
+            bands = (
+                "red",
+                "green",
+                "blue",
+                "nir",
+                "swir1",
+                "swir2",
+            )
+            if rgb_bands is None:
+                rgb_bands = ("red", "green", "blue")
+
+    @property
+    def measurements(self) -> Tuple[str, ...]:
+        return self.bands + self.aux_bands
+
+    def _native_tr(self, xx):
+        """
+        Loads in the data in the native projection. It performs the following:
+
+        1. Loads pq bands
+        2. Extract cloud_mask flags from bands
+        3. Add cloud_mask
+        4. Drops nodata pixels
+
+        .. bitmask::
+            15  14  13  12  11  10  9   8   7   6   5   4   3   2   1   0
+            |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   |
+            |   |   |   |   |   |   |   |   |   |   |   |   |   |   |   x-----> nodata
+            |   |   |   |   |   |   |   |   |   |   |   |   |   |   o---------> dilated_cloud
+            |   |   |   |   |   |   |   |   |   |   |   |   |   x-------------> cirrus
+            |   |   |   |   |   |   |   |   |   |   |   |   o-----------------> cloud
+            |   |   |   |   |   |   |   |   |   |   |   x---------------------> cloud_shadow
+            |   |   |   |   |   |   |   |   |   |   o-------------------------> snow
+            |   |   |   |   |   |   |   |   |   x-----------------------------> clear
+            |   |   |   |   |   |   |   |   o---------------------------------> water
+            |   |   |   |   |   |   |   x-------------------------------------> cloud_confidence
+            |   |   |   |   |   |   o-----------------------------------------> cloud_confidence
+            |   |   |   |   |   x---------------------------------------------> cloud_shadow_confidence
+            |   |   |   |   o-------------------------------------------------> cloud_shadow_confidence
+            |   |   |   x-----------------------------------------------------> snow_ice_confidence
+            |   |   o---------------------------------------------------------> snow_ice_confidence
+            |   x-------------------------------------------------------------> cirrus_confidence
+            0-----------------------------------------------------------------> cirrus_confidence
+        """
+        mask_band = xx[self.mask_band]
+        xx = xx.drop_vars([self.mask_band])
+
+        # set cloud_mask (cloud + cloud_shadow) bitmask - True=non-cloud, False=cloud
+        cloud_mask = da.bitwise_and(mask_band, 0b0000_0000_0001_1000) == 0
+        xx["cloud_mask"] = cloud_mask
+
+        # set no_data bitmask - True=data, False=no-data
+        keeps = da.bitwise_and(mask_band, 0b0000_0000_0000_0001) == 0
```

Comment:
I think this is our problem. We're doing this on a stack of data, right?

I thin we need to do this the other way. We should only drop where there are All nodata.

This is saying where it's never nodata, then keep it. But some scenes have nodata in different areas.

We want to say if it's ALWAYS nodata, then DROP IT.